runtime/manifest-parser.peg

```diff
@@ -293,7 +293,7 @@ RecipeParticle
 RecipeParticleItem = RecipeParticleConnection / RecipeParticleSlotConnection
 
 RecipeParticleConnection
-  = param:(lowerIdent / '*') whiteSpace dir:Direction whiteSpace target:ConnectionTarget eolWhiteSpace
+  = param:(lowerIdent / '*') whiteSpace dir:Direction whiteSpace? target:ConnectionTarget? eolWhiteSpace
```

Comment:
since whitespace is currently mandatory, this should be

```
target:(whitespace ConnectionTarget)?
```

then use the optional function to pull `target` back apart.